bd_spot_wrapper/spot_wrapper/spot.py

```diff
@@ -340,6 +350,79 @@ def get_image_responses(self, sources, quality=None, pixel_format=None):
 
         return image_responses
 
+    def setup_logging_sources(self, camera_sources: List[str]):
+        """
+        Order .. RGB & then DEPTH_IN_RGB
+        DO NOT SUPPORT DEPTH and RGB_IN_DEPTH.
+        """
+        source_list = []  # type: List[str]
+
+        if not camera_sources:
+            print("Empty list passed in logger camera sources")
+
+        # Verification checks
+        for camera_source in camera_sources:
+            if (
+                "_depth" in camera_source
+                and "_depth_in_visual_frame" not in camera_source
+                and "_depth_in_hand_color_frame" not in camera_source
+            ):
+                print(
+                    f"Invalid source name {camera_source}. Camera source should either be rgb or depth aligned in rgb"
+                )
+            else:
+                source_list.append(camera_source)
+
+        self.source_list = source_list  # TODO define in init
+        print(f"Initialized logging for sources : {self.source_list}")
+
+    def update_logging_data(self):
+
+        log_packet = {
+            "timestamp": time.time(),
+            "camera_data": [],
+            "vision_T_base": None,
+            "base_pose_xyt": None,
+            "arm_pose": None,
+        }  # type: Dict[str, Any]
+
+        for camera_source in self.source_list:
+            img_responses = self.get_image_responses([camera_source])
+            log_packet["camera_data"].append(
+                {
+                    "src_info": camera_source,
+                    "raw_image": image_response_to_cv2(
+                        img_responses[0]
+                    ),  # np.ndarray ... How to save? (Serialize to bytes / List / PNG)?
```

Comment:
I think png makes sense.
`from PIL import Image
depth_image = Image.fromarray(depth_array.astype('uint16'))
depth_image.save("depth_image.png")
 `
 something like this?

Reply:
Since output of `image_response_to_cv2` is an image as np.ndarray, we can store is as is in the pkl file.

So we dont need to store it as a separate file.